backend/functions/src/index.ts

```diff
@@ -1,62 +1,103 @@
 import { firestore, initializeApp } from "firebase-admin";
+import * as express from "express";
+import * as cors from "cors";
 
+// Types
+interface IResponse {
+  status: "success" | "fail" | "error";
+  data?: {};
+  message?: string;
+  code?: number;
+}
+
+// Init
 initializeApp();
-const cors = corsMod({ origin: true });
+const transactions = express();
+transactions.use(cors({ origin: true }));
+exports.transactions = functions
+  .region("europe-west1")
+  .https.onRequest(transactions);
 
 // Helper
-const getAllTransactions = async function () {
-  const snapshot = await firestore().collection("transactions").get();
-  return snapshot.docs.map((transaction) => ({
-    ...transaction.data(),
-    id: transaction.id,
-  }));
+const docWithID = function (
+  docSnapshot: firestore.DocumentSnapshot<firestore.DocumentData>,
+) {
+  return {
+    ...docSnapshot.data(),
+    id: docSnapshot.id,
+  };
 };
 
-// addTransaction
-exports.addTransaction = functions
-  .region("europe-west1")
-  .https.onRequest(async (req, res) => {
-    cors(req, res, async () => {
-      // ??? improve with JSON post?
-      const transaction = req.query.transaction
-        ? JSON.parse(req.query.transaction as string)
-        : {};
+// GET /transactions/
+transactions.get("/", (req, res) => {
+  firestore()
+    .collection("transactions")
+    .get()
+    .then((docSnapshots) => {
+      const data = docSnapshots.docs.map((docSnapshot) =>
+        docWithID(docSnapshot),
+      );
 
-      await firestore().collection("transactions").add(transaction);
+      const response: IResponse = {
+        status: "success",
+        data: { transactions: data },
+      };
+      res.json(response);
+    });
+});
+
+// GET /transactions/:id
+// ??? Error handling: doc not found
+transactions.get("/:id", (req, res) => {
+  const id = req.params.id;
 
-      const data = await getAllTransactions();
-      res.json({ data });
+  firestore()
+    .doc(`transactions/${id}`)
+    .get()
+    .then((docSnapshot) => {
+      const response: IResponse = {
+        status: "success",
+        data: { transaction: docWithID(docSnapshot) },
+      };
+      res.json(response);
     });
-  });
+});
 
-// deleteTransactions
-exports.deleteTransactions = functions
-  .region("europe-west1")
-  .https.onRequest(async (req, res) => {
-    cors(req, res, async () => {
-      // ??? improve with JSON post?
-      const ids = req.query.ids ? JSON.parse(req.query.ids as string) : [];
-      const writeBatch = firestore().batch();
-
-      ids.forEach((id: string) => {
-        writeBatch.delete(firestore().doc(`transactions/${id}`));
-        functions.logger.info("added to batch: ", id);
-      });
-      await writeBatch.commit();
-
-      const data = await getAllTransactions();
-      res.json({ data });
+// POST /transactions/
+// ??? Error handling: failed to create
+transactions.post("/", (req, res) => {
+  const transaction = req.body.data;
+
+  firestore()
```

Comment:
Il y a une raison (autre que comprendre les cloud functions) pour faire tes modifications dans des functions plutôt que directement sur le client?

Reply:
Cloud functions replaced by Firestore front-end SDK in PR #11